src/som/interpreter/actors/Actor.java

```diff
@@ -231,15 +239,20 @@ public void doRunWithObjectSafepoints(final ActorProcessingThread t) {
 
       t.currentlyExecutingActor = actor;
 
-      if (VmSettings.ACTOR_TRACING) {
-        ActorExecutionTrace.recordActivityContext(actor, tracer);
+      if (VmSettings.UNIFORM_TRACING) {
+        UniformExecutionTrace.recordActivityContext(actor, tracer);
       } else if (VmSettings.KOMPOS_TRACING) {
         KomposTrace.currentActivity(actor);
       }
 
       while (getCurrentMessagesOrCompleteExecution()) {
         processCurrentMessages(t, dbg);
       }
+
+      if (VmSettings.UNIFORM_TRACING || VmSettings.KOMPOS_TRACING) {
+        t.swapTracingBufferIfRequestedUnsync();
+      }
+      t.currentlyExecutingActor = null;
```

Comment:
```suggestion
```

I think we to make `doRunWithObjectSafepoints` `private` and the `finally clause of `doRun` already does this.
What do you think?